lib/bridge-client/index.js

```diff
@@ -528,32 +628,37 @@ BridgeClient.prototype._handleShardTmpFileFinish = function(state, meta, done) {
   var self = this;
   var hash = utils.rmd160(meta.hash);
   var auditGenerator = new AuditStream(3);
-  var shardFile = fs.createReadStream(meta.tmpName);
+
+  var rs = self._store.createReadStream(meta.tmpName);
 
   self._logger.info('Hash for this shard is: %s', hash);
 
   function _handleError(err) {
     self._logger.warn('Failed to upload shard...');
-    state.cleanup();
-    return state.callback(err);
+    state.cleanup(function(err2){
```

Comment:
Note though, an error here means that we were unable to remove a file from a blobstore for whatever reason.